### PR TITLE
feat: move to dotnet 8.0.2

### PR DESCRIPTION
**Pull request by emalfroy**
BREAKING CHANGE: move to dotnet 8.0.2

### CODE REVIEW

File: paket.dependencies [hunk 1]
```diff
@@ -1,114 +1,110 @@
...
-nuget Microsoft.Extensions.Configuration.Json 6.0.0
-nuget Microsoft.Extensions.DependencyInjection 6.0.0
-nuget Microsoft.Extensions.DependencyInjection.Abstractions 7.0.0
+nuget Microsoft.Extensions.Configuration.Binder 8.0.1
+nuget Microsoft.Extensions.Configuration.Json 8.0.0
+nuget Microsoft.Extensions.DependencyInjection 8.0.0
+nuget Microsoft.Extensions.DependencyInjection.Abstractions 8.0.1
 
 // .NET CLASSIC STUFF
 nuget EntityFramework 6.4.4
```
Comment (jvandaal) [inline paket.dependencies:110]: EntityFramework can be removed.

File: paket.dependencies [hunk 2]
```diff
@@ -1,114 +1,110 @@
-framework: net6.0, net472
+framework: net8.0, net472
```
Comment (jvandaal) [inline paket.dependencies:1]: Remove net472

File: src/AddressRegistry.Producer.Snapshot.Oslo/AddressRegistry.Producer.Snapshot.Oslo.csproj [hunk 3]
```diff
@@ -9,12 +9,12 @@
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
```
Comment (jvandaal) [inline src/AddressRegistry.Producer.Snapshot.Oslo/AddressRegistry.Producer.Snapshot.Oslo.csproj:11]: Entire PropertyGroup can be removed

File: src/AddressRegistry.Producer/AddressRegistry.Producer.csproj [hunk 4]
```diff
@@ -7,12 +7,12 @@
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
```
Comment (jvandaal) [inline src/AddressRegistry.Producer/AddressRegistry.Producer.csproj:9]: Entire PropertyGroup can be removed

File: src/AddressRegistry.Projector/AddressRegistry.Projector.csproj [hunk 5]
```diff
@@ -9,12 +9,12 @@
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
```
Comment (jvandaal) [inline src/AddressRegistry.Projector/AddressRegistry.Projector.csproj:11]: Entire PropertyGroup can be removed

File: src/EF.MigrationsHelper/EF.MigrationHelper.csproj [hunk 6]
```diff
@@ -1,10 +1,9 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
     <LangVersion>latest</LangVersion>
 
-    <RuntimeFrameworkVersion>6.0.3</RuntimeFrameworkVersion>
```
Comment (jvandaal) [inline src/EF.MigrationsHelper/EF.MigrationHelper.csproj:7]: I think this should be kept.

Comment (jvandaal) [reply src/EF.MigrationsHelper/EF.MigrationHelper.csproj:7]: But change the version

File: src/AddressRegistry.Api.BackOffice.Handlers.Lambda/AddressRegistry.Api.BackOffice.Handlers.Lambda.csproj [hunk 7]
```diff
@@ -14,6 +14,7 @@
 
     <!-- Error NETSDK1152: https://docs.microsoft.com/en-us/dotnet/core/compatibility/sdk/6.0/duplicate-files-in-output -->
     <ErrorOnDuplicatePublishOutputFiles>false</ErrorOnDuplicatePublishOutputFiles>
+    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
```
Comment (jvandaal) [inline src/AddressRegistry.Api.BackOffice.Handlers.Lambda/AddressRegistry.Api.BackOffice.Handlers.Lambda.csproj:17]: Why is this added?

Comment (jvandaal) [reply src/AddressRegistry.Api.BackOffice.Handlers.Lambda/AddressRegistry.Api.BackOffice.Handlers.Lambda.csproj:17]: This is missing https://github.com/Informatievlaanderen/building-registry/blob/main/src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/BuildingRegistry.Api.BackOffice.Handlers.Lambda.csproj#L16

Comment (jvandaal) [reply src/AddressRegistry.Api.BackOffice.Handlers.Lambda/AddressRegistry.Api.BackOffice.Handlers.Lambda.csproj:17]: aws-lambda-tools-defaults.json should also be updated

File: src/AddressRegistry.Api.BackOffice/Infrastructure/Startup.cs [hunk 8]
```diff
@@ -3,18 +3,18 @@ namespace AddressRegistry.Api.BackOffice.Infrastructure
...
     using System.Reflection;
+    using Asp.Versioning.ApiExplorer;
     using Autofac;
     using Autofac.Extensions.DependencyInjection;
     using Be.Vlaanderen.Basisregisters.Auth.AcmIdm;
     using Be.Vlaanderen.Basisregisters.AggregateSource.SqlStreamStore;
     using Be.Vlaanderen.Basisregisters.Api;
     using Be.Vlaanderen.Basisregisters.CommandHandling.Idempotency;
-    using Be.Vlaanderen.Basisregisters.DataDog.Tracing.Microsoft;
+    using Be.Vlaanderen.Basisregisters.DataDog.Tracing.Autofac;
```
Comment (jvandaal) [inline src/AddressRegistry.Api.BackOffice/Infrastructure/Startup.cs:13]: I'm missing this change
https://github.com/Informatievlaanderen/building-registry/blob/main/src/BuildingRegistry.Api.BackOffice/Infrastructure/Startup.cs#L94

Comment (jvandaal) [reply src/AddressRegistry.Api.BackOffice/Infrastructure/Startup.cs:13]: Then you'll need to add https://github.com/Informatievlaanderen/building-registry/blob/main/src/BuildingRegistry.Api.BackOffice/Infrastructure/Modules/FluentValidationModule.cs

File: .github/workflows/build.yml [hunk 9]
```diff
@@ -53,15 +54,15 @@ jobs:
...
+      uses: actions/cache@v4
       with:
         path: ~/sonar/cache
         key: ${{ runner.os }}-sonar
         restore-keys: ${{ runner.os }}-sonar
         
     - name: Cache SonarCloud scanner
       id: cache-sonar-scanner
-      uses: actions/cache@v1
+      uses: actions/cache@v4
```
Comment (jvandaal) [inline .github/workflows/build.yml:65]: Also change restore packages to
        dotnet tool restore
        dotnet paket restore
        dotnet restore

File: .github/workflows/release.yml [hunk 10]
```diff
@@ -55,314 +57,320 @@ jobs:
...
       test-project: AddressRegistry.Tests
       build-project: AddressRegistry.Producer.Snapshot.Oslo
       semver: ${{ needs.set-release-version.outputs.version }}
+      dotnet-version: ${{ vars.VBR_DOTNET_VERSION_8 }}
     secrets: inherit
 
   build-migrator-address:
     name: Build Migrator Address
-    uses: Informatievlaanderen/build-pipeline/.github/workflows/build-image.yml@v6.2.9
+    uses: Informatievlaanderen/build-pipeline/.github/workflows/build-image.yml@main
```
Comment (jvandaal) [inline .github/workflows/release.yml:255]: Migrator can be removed. Also remove all related steps below.

File: .github/workflows/release.yml [hunk 11]
```diff
@@ -415,10 +423,10 @@ jobs:
       shell: bash
 
     - name: Setup Node.js
-      uses: actions/setup-node@v3
+      uses: actions/setup-node@v4
```
Comment (jvandaal) [inline .github/workflows/release.yml:426]: Add
```
 with:
        node-version: '20.x'
```

File: .github/workflows/release.yml [hunk 12]
```diff
@@ -722,18 +730,18 @@ jobs:
...
           aws-region: ${{ secrets.VBR_AWS_REGION_PRD }}
 
       - name: Login to Amazon ECR
-        uses: aws-actions/amazon-ecr-login@v1.5.3
+        uses: aws-actions/amazon-ecr-login@v2
 
       # Download artifact
       - name: Download artifact
-        uses: actions/download-artifact@v3
+        uses: actions/download-artifact@v4
```
Comment (jvandaal) [inline .github/workflows/release.yml:744]: Remove docker tag when pushing images for the first time. `BUILD_DOCKER_REGISTRY_TST` can also be removed in this step

File: .github/workflows/release.yml [hunk 13]
```diff
@@ -788,18 +796,18 @@ jobs:
...
           aws-region: ${{ secrets.VBR_AWS_REGION_PRD }}
 
       - name: Login to Amazon ECR (Staging)
-        uses: aws-actions/amazon-ecr-login@v1.5.3
+        uses: aws-actions/amazon-ecr-login@v4
 
       # Download artifact
       - name: Download artifact
-        uses: actions/download-artifact@v3
+        uses: actions/download-artifact@v4
```
Comment (jvandaal) [inline .github/workflows/release.yml:810]: When pushing to staging, change `BUILD_DOCKER_REGISTRY_TST: ${{ secrets.VBR_DEVOPS_DOCKER_REGISTRY }}`

File: .github/workflows/release.yml [hunk 14]
```diff
@@ -853,18 +861,18 @@ jobs:
...
           aws-region: ${{ secrets.VBR_AWS_REGION_PRD }}
 
       - name: Login to Amazon ECR (New Production)
-        uses: aws-actions/amazon-ecr-login@v1.5.3
+        uses: aws-actions/amazon-ecr-login@v2
 
       # Download artifact
       - name: Download artifact
-        uses: actions/download-artifact@v3
+        uses: actions/download-artifact@v4
```
Comment (jvandaal) [inline .github/workflows/release.yml:875]: When pushing to newproduction change `BUILD_DOCKER_REGISTRY_TST: ${{ secrets.VBR_DEVOPS_DOCKER_REGISTRY }}`

File: src/AddressRegistry.Api.BackOffice/AddressRegistry.Api.BackOffice.csproj [hunk 15]
```diff
@@ -10,12 +10,12 @@
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
-    <DocumentationFile>bin\Debug\net6.0\AddressRegistry.Api.BackOffice.xml</DocumentationFile>
+    <DocumentationFile>bin\Debug\net8.0\AddressRegistry.Api.BackOffice.xml</DocumentationFile>
```
Comment (jvandaal) [inline src/AddressRegistry.Api.BackOffice/AddressRegistry.Api.BackOffice.csproj:13]: paket.template of api.backoffice is still on net6.0

File: paket.dependencies [hunk 16]
```diff
@@ -1,117 +1,112 @@
...
-nuget AspNetCore.HealthChecks.SqlServer 6.0.2
-nuget AspNetCore.HealthChecks.NpgSql 6.0.2
+nuget AspNetCore.HealthChecks.SqlServer 8.0.0
+nuget AspNetCore.HealthChecks.NpgSql 8.0.0
 
 nuget AsyncEnumerator 4.0.2
-nuget Dapper 2.0.123
-nuget CsvHelper 27.2.1
+nuget Dapper 2.1.35
+nuget CsvHelper 31.0.2
```
Comment (ArneD) [inline paket.dependencies:12]: can you check if we still need this?

File: paket.dependencies [hunk 17]
```diff
@@ -1,117 +1,112 @@
...
-nuget Npgsql.EntityFrameworkCore.PostgreSQL 6.0.3
+nuget Npgsql 8.0.2
+nuget Npgsql.EntityFrameworkCore.PostgreSQL 8.0.2
 nuget Npgsql.EntityFrameworkCore.PostgreSQL.Design 2.0.0-preview1
-nuget Npgsql.EntityFrameworkCore.PostgreSQL.NetTopologySuite 6.0.3
+nuget Npgsql.EntityFrameworkCore.PostgreSQL.NetTopologySuite 8.0.2
 
-nuget RestSharp 107.3.0
+nuget RestSharp 110.2.0
 nuget GeoJSON.NET 1.2.19
```
Comment (ArneD) [inline paket.dependencies:29]: Can you check if we still need these 2?

File: paket.dependencies [hunk 18]
```diff
@@ -1,117 +1,112 @@
...
 // LAMBDA STUFF
 
-nuget Microsoft.Extensions.Configuration.Binder 6.0.0
-nuget Microsoft.Extensions.Configuration.Json 6.0.0
-nuget Microsoft.Extensions.DependencyInjection 6.0.0
-nuget Microsoft.Extensions.DependencyInjection.Abstractions 7.0.0
+nuget Microsoft.Extensions.Configuration.Binder 8.0.1
+nuget Microsoft.Extensions.Configuration.Json 8.0.0
+nuget Microsoft.Extensions.DependencyInjection 8.0.0
+nuget Microsoft.Extensions.DependencyInjection.Abstractions 8.0.1
```
Comment (ArneD) [inline paket.dependencies:107]: Can this be removed or not? Same as the first lines

File: src/AddressRegistry.CacheWarmer/Dockerfile [hunk 19]
```diff
@@ -1,5 +1,4 @@
 FROM alpine:3.10.3
```
Comment (ArneD) [inline src/AddressRegistry.CacheWarmer/Dockerfile:1]: upgrade to 3.19.1

File: paket.dependencies (hunk 16 above)
Comment (emalfroy) [reply paket.dependencies:12]: is used in AddressRegistry.Importer.AddressMatch
